### PR TITLE
375 db: migrations to harmonize prod db with the code

### PR DESCRIPTION
**Pull request by turtle321**

### CODE REVIEW

File: scripts/list_all_tables_and_indexes.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ORDER BY constraint_name;
+"""
+
+
+def list_tables_and_indexes():
+    """
+    List all tables and indexes (including public keys).
+
+    Returns:
+        dict(set(strings))
```
[the remark this answers is not in the document]
Comment (turtle321) [reply scripts/list_all_tables_and_indexes.py]: Done. But I do not want to use proper typing in Python. Also I support lightweight guidelines for docstrings.

File: scripts/list_all_tables_and_indexes.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    """
+    # Get all tables.
+    tables = db.engine.execute(SELECT_ALL_TABLES).fetchall()
+
+    result = {}
+    for table in tables:
+        table_name = str(table[1])
+
+        # Get all indexes.
+        indexes = db.engine.execute(SELECT_INDEXES.format(table_name)).fetchall()
```
[the remark this answers is not in the document]
Comment (turtle321) [reply scripts/list_all_tables_and_indexes.py]: Done

File: scripts/list_all_tables_and_indexes.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (michamos) [reply scripts/list_all_tables_and_indexes.py]: I don't really care here, but I disagree in general. One has a clear meaning, even understandable by a type-checker such as `mypy`. What does your example even mean? a `dict` is a mapping from one thing (a key) to another (a value). So you need two types to parameterize a `dict`.

File: inspirehep/alembic/53e8594bc789_fix_the_production_db.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+
+def upgrade():
+    # Rename indexes.
+    for _, names in PRIMARY_KEYS_MAP:
+        op.execute('ALTER INDEX {} RENAME TO {}'.format(names['old'], names['new']))
+
+    # Create indexes.
+    op.execute("CREATE INDEX CONCURRENTLY IF NOT EXISTS json_ids_index ON records_metadata USING gin ((json -> 'ids'))")
```
[the remark this answers is not in the document]
Comment (turtle321) [reply inspirehep/alembic/53e8594bc789_fix_the_production_db.py]: Done

File: inspirehep/alembic/53e8594bc789_fix_the_production_db.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+     'transaction': dict(old='transaction_pkey', new='pk_transaction'),
+     'userprofiles_userprofile': dict(old='userprofiles_userprofile_pkey', new='pk_userprofiles_userprofile'),
+     'workflows_audit_logging': dict(old='workflows_audit_logging_pkey', new='pk_workflows_audit_logging'),
+     'workflows_buckets': dict(old='workflows_buckets_pkey', new='pk_workflows_buckets'),
+     'workflows_object': dict(old='workflows_object_pkey', new='pk_workflows_object'),
+     'workflows_pending_record': dict(old='workflows_pending_record_pkey', new='pk_workflows_pending_record'),
+     'workflows_workflow': dict(old='workflows_workflow_pkey', new='pk_workflows_workflow'),
+}
+
+INDEXES_MAP = {
```
[the remark this answers is not in the document]
Comment (turtle321) [reply inspirehep/alembic/53e8594bc789_fix_the_production_db.py]: I forgot, fixed now.

**Comment by turtle321** [comment]
ci merge please

**Comment by turtle321** [comment]
ci merge please

**Pull request merged**